indexing-service/src/main/java/io/druid/indexing/overlord/TaskStorage.java

```diff
@@ -53,6 +53,13 @@
   public void addLock(String taskid, TaskLock taskLock);
 
   /**
+   * Upgrades the lock state in the storage facility to non-preemptive lock
```

Comment:
yes that's correct. I wanted to keep the return void just like `addLock` but it makes sense to throw IllegalArgumentException